contracts/token/ERC721/ERC721Metadata.sol

```diff
@@ -56,7 +59,16 @@ contract ERC721Metadata is Context, ERC165, ERC721, IERC721Metadata {
      * Throws if the token ID does not exist. May return an empty string.
      * @param tokenId uint256 ID of the token to query
      */
-    function tokenURI(uint256 tokenId) external view returns (string memory) {
+    function tokenURI(uint256 tokenId) public view returns (string memory) {
+        return string(abi.encodePacked(_baseTokenURI(), _tokenURI(tokenId)));
+    }
+
+    /**
+     * @dev Internal returns an URI for a given token ID.
+     * Throws if the token ID does not exist. May return an empty string.
+     * @param tokenId uint256 ID of the token to query
+     */
+    function _tokenURI(uint256 tokenId) internal view returns (string memory) {
```

Comment:
I get the intent behind trying to get the actual token URI (without the base), but we should name this something else: it's very confusing that `tokenURI()` and `_tokenURI()` return different things.

Reply:
I see your point. 

What do you think of `_getTokenURI`; `_invidualTokenURI`; `_singularTokenURI`?  any idea?

If we can't find a good name, we can try with changing the developer notation.